client/state/purchases/actions.js

```diff
@@ -13,15 +13,19 @@ import {
 	PURCHASE_REMOVE_FAILED,
 } from 'calypso/state/action-types';
 import { requestHappychatEligibility } from 'calypso/state/happychat/user/actions';
-
+import { requestAdminMenu } from 'calypso/state/admin-menu/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import 'calypso/state/purchases/init';
 
 const PURCHASES_FETCH_ERROR_MESSAGE = i18n.translate( 'There was an error retrieving purchases.' );
 const PURCHASE_REMOVE_ERROR_MESSAGE = i18n.translate( 'There was an error removing the purchase.' );
 
-export const clearPurchases = () => ( dispatch ) => {
+export const clearPurchases = () => ( dispatch, getState ) => {
+	const siteId = getSelectedSiteId( getState() );
+
 	dispatch( { type: PURCHASES_REMOVE } );
 	dispatch( requestHappychatEligibility() );
+	dispatch( requestAdminMenu( siteId ) );
```

Comment:
Can we add a safeguard check for cases where purchases are cleared but no site is selected?

```suggestion
	if ( siteId ) {
		dispatch( requestAdminMenu( siteId ) );
	}
```